src/SimplerSoftware.EntityFrameworkCore.SqlServer.NodaTime/build/netstandard2.0/SimplerSoftware.EntityFrameworkCore.SqlServer.NodaTime.targets

```
@@ -0,0 +1,46 @@
+﻿<Project>
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+    <EFCoreSqlServerNodaTimeFile>$(IntermediateOutputPath)EFCoreSqlServerNodaTime$(DefaultLanguageSourceExtension)</EFCoreSqlServerNodaTimeFile>
+  </PropertyGroup>
+  <Choose>
+    <When Condition="'$(Language)' == 'F#'">
+      <Choose>
+        <When Condition="'$(OutputType)' == 'Exe' OR '$(OutputType)' == 'WinExe'">
+          <PropertyGroup>
+            <CodeFragmentItemGroup>CompileBefore</CodeFragmentItemGroup>
+          </PropertyGroup>
+        </When>
+        <Otherwise>
+          <PropertyGroup>
+            <CodeFragmentItemGroup>CompileAfter</CodeFragmentItemGroup>
+          </PropertyGroup>
+        </Otherwise>
+      </Choose>
+    </When>
+    <Otherwise>
+      <PropertyGroup>
+        <CodeFragmentItemGroup>Compile</CodeFragmentItemGroup>
+      </PropertyGroup>
+    </Otherwise>
+  </Choose>
+  <Target Name="AddEFCoreSqlServerNodaTime"
+          BeforeTargets="CoreCompile"
+          DependsOnTargets="PrepareForBuild"
+          Condition="'$(DesignTimeBuild)' != 'True'"
+          Inputs="$(MSBuildAllProjects)"
+          Outputs="$(EFCoreSqlServerNodaTimeFile)">
+    <ItemGroup>
+      <EFCoreSqlServerNodaTimeServices Include="Microsoft.EntityFrameworkCore.Design.DesignTimeServicesReferenceAttribute">
+        <_Parameter1>SimplerSoftware.EntityFrameworkCore.SqlServer.NodaTime.Design.SqlServerNodaTimeDesignTimeServices, SimplerSoftware.EntityFrameworkCore.SqlServer.NodaTime</_Parameter1>
+        <_Parameter2>SimplerSoftware.EntityFrameworkCore.SqlServer</_Parameter2>
```

```suggestion
        <_Parameter2>Microsoft.EntityFrameworkCore.SqlServer</_Parameter2>
```

(This is the *provider* the IDesignTimeServices should be used for)

---

Note, this is the "magic" Erik is talking about. When you install this package, it adds an assembly-level attribute to the app that the EF Core tool will search for and use to load additional design-time services.